--- a/config/config.h	2018-09-30 13:57:54.659223156 -0400
+++ b/config/config.h	2018-09-30 13:58:23.675087718 -0400
@@ -60,7 +60,7 @@
 
 /* Define to 1 if your system has a GNU libc compatible `malloc' function, and
    to 0 otherwise. */
-#define HAVE_MALLOC 0
+#define HAVE_MALLOC 1
 
 /* Define to 1 if you have the <math.h> header file. */
 #define HAVE_MATH_H 1
@@ -238,7 +238,7 @@
 #endif
 
 /* Define to rpl_malloc if the replacement function should be used. */
-#define malloc rpl_malloc
+/* #define malloc rpl_malloc */
 
 /* Define to `long int' if <sys/types.h> does not define. */
 /* #undef off_t */
